mapping/mapping_vectors.go

```diff
@@ -0,0 +1,119 @@
+//  Copyright (c) 2023 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build vectors
+// +build vectors
+
+package mapping
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/blevesearch/bleve/v2/document"
+	"github.com/blevesearch/bleve/v2/util"
+)
+
+func NewVectorFieldMapping() *FieldMapping {
+	return &FieldMapping{
+		Type:         "vector",
+		Store:        false,
+		Index:        true,
+		IncludeInAll: false,
+		DocValues:    false,
+		SkipFreqNorm: true,
+	}
+}
+
+func (fm *FieldMapping) processVector(propertyMightBeVector interface{},
+	pathString string, path []string, indexes []uint64, context *walkContext) {
+	propertyVal := reflect.ValueOf(propertyMightBeVector)
+	if !propertyVal.IsValid() {
+		return
+	}
+
+	// Validating the length of the vector is required here, in order to
+	// help zapx in deciding the shape of the batch of vectors to be indexed.
+	if propertyVal.Kind() == reflect.Slice && propertyVal.Len() == fm.Dims {
+		vector := make([]float32, propertyVal.Len())
+		isVectorValid := true
+		for i := 0; i < propertyVal.Len(); i++ {
```

Comment:
This is going to be a major bottle neck while indexing vectors - I think we can rely on faiss to index or silently drop a vector whose content is mangled, as long as the length matches `Dims`. Is it possible for us to not have to iterate and assert but simply pass in the []interface{} ?

Reply:
@moshaad7 something like this - https://go.dev/play/p/Vr5tCw81hSf

Reply:
Tried benchmarking this approach with a vector of dimension `768`, turns out the JSON unmarshal/marshal maneuver is way more costlier. 

```
BenchmarkInterpretEachFloat32-10    	  294214	      3480 ns/op	    3072 B/op	       1 allocs/op
BenchmarkInterpretJSON-10           	    6541	    182118 ns/op	   36840 B/op	     803 allocs/op
```

Will rest this case for now.